MGDOBR-1133: Fix timeouts

sjamboro opened the pull request.

sjamboro commented on cypress/integration/smoke/RESTInteractionTest.ts:
```diff
@@ -109,7 +109,7 @@ onlyOn(isEnvironmentType(EnvType.Dev), () => {
           cy.get("td:first").should("have.text", bridgeName);
           cy.get("td:nth-child(3)").then(($state) => {
-            cy.wrap($state, { timeout: 60000 }).should("have.text", "Ready");
+            cy.wrap($state, { timeout: 240000 }).should("have.text", "Ready");
```
This timeout is similar like in https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/integration-tests/src/test/resources/features/bridge.feature

kelvah replied:
Should we also increase timeouts during deletion checks? See [comment on the other PR](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/226#issuecomment-1281977401)

jomarko commented on cypress/support/commands.ts:
```diff
@@ -62,15 +62,6 @@ Cypress.Commands.add(
 );
 
 Cypress.Commands.add("login", () => {
-  cy.wait(4000); //prevent random failure
-  cy.get("#rh-password-verification-submit-button").then(($item) => {
-    if ($item.is(":visible")) {
-      cy.log("Cookie's choise is not present");
-    } else {
-      cy.log("Cookie's choise is present");
-      cy.get("button:contains('Accept default')").click().should("not.exist");
-    }
-  });
```
I agree with removing this part of login command. 

I was not able to find out an element, that would match `#rh-password-verification-submit-button` selector. To me it seems like an outdated?

kelvah replied:
I thought this part was responsible for closing the cookie pop-up. Is it covered elsewhere now? Or isn't it necessary anymore?

Sorry, I am kinda lost. This PR title and description don't help much understanding what's going on.

sjamboro replied:
@kelvah Unfortunately it is not cover right now. I can not reproduce it locally so I have not got the html code. The condition does not work properly. The problem still occurs in GHAs.

kelvah replied:
@sjamboro Thanks for the explanation.

sjamboro commented:
**Run #0 - Results: 1/9 failed** - The screenshot shows the new instance but the popover is missing:
```
<failure message="Timed out retrying after 120000ms: `cy.should()` failed because this element is detached from the DOM. `<div data-ouia-component-id="steps-count" data-ouia-component-type="QE/StackItem" class="pf-l-stack__item">1 of 3 ...</div>` Cypress requires elements be attached in the DOM to interact with them. The previous command that ran was: > `cy.ouiaId()` This DOM element likely became detached somewhere between the previous and current command. Common situations why this happens: - Your JS framework re-rendered asynchronously - Your app code reacted to an event firing and removed the element You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands. https://on.cypress.io/element-has-detached-from-dom" type="CypressError">
<![CDATA[ CypressError: Timed out retrying after 120000ms: `cy.should()` failed because this element is detached from the DOM. `<div data-ouia-component-id="steps-count" data-ouia-component-type="QE/StackItem" class="pf-l-stack__item">1 of 3 ...</div>` Cypress requires elements be attached in the DOM to interact with them. The previous command that ran was: > `cy.ouiaId()` This DOM element likely became detached somewhere between the previous and current command. Common situations why this happens: - Your JS framework re-rendered asynchronously - Your app code reacted to an event firing and removed the element You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands. https://on.cypress.io/element-has-detached-from-dom at $Cy.ensureAttached
(webpack:///./integration/instance/InstanceCreation.ts:68:70) ]]>
</failure>
```

sjamboro commented:
**Run #1 - Results: 1/9 failed** - The screenshot shows the new instance but the popover is missing:
```
<failure message="Timed out retrying after 120000ms: `cy.should()` failed because this element is detached from the DOM. `<div data-ouia-component-id="steps-count" data-ouia-component-type="QE/StackItem" class="pf-l-stack__item">1 of 3 ...</div>` Cypress requires elements be attached in the DOM to interact with them. The previous command that ran was: > `cy.ouiaId()` This DOM element likely became detached somewhere between the previous and current command. Common situations why this happens: - Your JS framework re-rendered asynchronously - Your app code reacted to an event firing and removed the element You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands. https://on.cypress.io/element-has-detached-from-dom" type="CypressError">
<![CDATA[ CypressError: Timed out retrying after 120000ms: `cy.should()` failed because this element is detached from the DOM. `<div data-ouia-component-id="steps-count" data-ouia-component-type="QE/StackItem" class="pf-l-stack__item">1 of 3 ...</div>` Cypress requires elements be attached in the DOM to interact with them. The previous command that ran was: > `cy.ouiaId()` This DOM element likely became detached somewhere between the previous and current command. Common situations why this happens: - Your JS framework re-rendered asynchronously - Your app code reacted to an event firing and removed the element You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands. https://on.cypress.io/element-has-detached-from-dom at $Cy.ensureAttached
(webpack:///./integration/instance/InstanceCreation.ts:68:70) ]]>
</failure>
```

sjamboro commented:
**Run `#2` - Results: 1/9 failed** - The screenshot shows the new instance but the popover is missing:
```
<failure message="Timed out retrying after 120000ms: `cy.should()` failed because this element is detached from the DOM. `<div data-ouia-component-id="steps-count" data-ouia-component-type="QE/StackItem" class="pf-l-stack__item">1 of 3 ...</div>` Cypress requires elements be attached in the DOM to interact with them. The previous command that ran was: > `cy.ouiaId()` This DOM element likely became detached somewhere between the previous and current command. Common situations why this happens: - Your JS framework re-rendered asynchronously - Your app code reacted to an event firing and removed the element You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands. https://on.cypress.io/element-has-detached-from-dom" type="CypressError">
<![CDATA[ CypressError: Timed out retrying after 120000ms: `cy.should()` failed because this element is detached from the DOM. `<div data-ouia-component-id="steps-count" data-ouia-component-type="QE/StackItem" class="pf-l-stack__item">1 of 3 ...</div>` Cypress requires elements be attached in the DOM to interact with them. The previous command that ran was: > `cy.ouiaId()` This DOM element likely became detached somewhere between the previous and current command. Common situations why this happens: - Your JS framework re-rendered asynchronously - Your app code reacted to an event firing and removed the element You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands. https://on.cypress.io/element-has-detached-from-dom at $Cy.ensureAttached
(webpack:///./integration/instance/InstanceCreation.ts:68:70) ]]>
</failure>
```

sjamboro commented:
**Run `#3` - Results: Passed**

sjamboro commented:
**Run `#4` - Results: 4/9 failed** - GDPR forms is present in all cases
```
<failure message="Timed out retrying after 10000ms: Expected to find element: `button:contains('Accept default')`, but never found it. Because this error occurred during a `before each` hook we are skipping the remaining tests in the current suite: `the 'Create a SE instance' ...`" type="AssertionError">
<![CDATA[ AssertionError: Timed out retrying after 10000ms: Expected to find element: `button:contains('Accept default')`, but never found it. Because this error occurred during a `before each` hook we are skipping the remaining tests in the current suite: `the 'Create a SE instance' ...` at Context.eval (webpack:///./support/commands.ts:71:11) ]]>
</failure>
```

jomarko commented on cypress/support/commands.ts:
```diff
@@ -62,15 +62,17 @@ Cypress.Commands.add(
-    }
-  });
+  //https://docs.cypress.io/guides/core-concepts/conditional-testing#Element-existence
+  cy.get("body", { timeout: 10000 })
+    .should("be.visible")
+    .then(($body) => {
+      if ($body.find('div[class="truste_overlay"]').length) {
+        // the form was found, do something else here
+        cy.get("button:contains('Accept default')").click().should("not.exist");
+      } else {
+        cy.log("Cookie's choise is not present");
+      }
+    });
```
I am approving this snippet without further checking. I struggle to get cookies dialogue locally so it is just my trust that locators are correct. However as @sjamboro wrote, it doesn't seems to be like 'permanent' solution

sjamboro commented:
@jomarko @kelvah I want to reverse this commit: https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/pull/223/commits/5abb2f3797cdde462597c9a1ab6e126e1a77bab6

It does not prevent test to fail (run # 0,1,2,5):
```
<failure message="Timed out retrying after 120000ms: `cy.should()` failed because this element is detached from the DOM. `<div data-ouia-component-id="steps-count" data-ouia-component-type="QE/StackItem" class="pf-l-stack__item">1 of 3 ...</div>` Cypress requires elements be attached in the DOM to interact with them. The previous command that ran was: > `cy.wrap()` This DOM element likely became detached somewhere between the previous and current command. Common situations why this happens: - Your JS framework re-rendered asynchronously - Your app code reacted to an event firing and removed the element You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands. https://on.cypress.io/element-has-detached-from-dom" type="CypressError">
<![CDATA[ CypressError: Timed out retrying after 120000ms: `cy.should()` failed because this element is detached from the DOM. `<div data-ouia-component-id="steps-count" data-ouia-component-type="QE/StackItem" class="pf-l-stack__item">1 of 3 ...</div>` Cypress requires elements be attached in the DOM to interact with them. The previous command that ran was: > `cy.wrap()` This DOM element likely became detached somewhere between the previous and current command. Common situations why this happens: - Your JS framework re-rendered asynchronously - Your app code reacted to an event firing and removed the element You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands. https://on.cypress.io/element-has-detached-from-dom at $Cy.ensureAttached (
(https://prod.foo.redhat.com:1337/__cypress/runner/cypress_runner.js:5904:14) From Your Spec Code: at Context.eval 
(webpack:///./integration/instance/InstanceCreation.ts:72:13) ]]>
</failure>
```

sjamboro commented:
Today, I have run quality checks twice and in both cases they succeeded.
